src/plugins/ui_feature/selection_input.ts

```diff
@@ -182,13 +182,13 @@ export class SelectionInputPlugin extends UIPlugin implements StreamCallbacks<Se
    * Insert new inputs after the given index.
    */
   private insertNewRange(index: number, values: string[]) {
-    const currentMaxId = Math.max(0, ...this.ranges.map((range) => Number(range.id)));
+    const currentMaxId = Math.max(-1, ...this.ranges.map((range) => Number(range.id)));
```

Comment:
The fact that id started with 1 instaed of 0 was not an issue, the problem was that when you changed the behaviour, you added a +1 to the counter of colors, and I do'nt ythink it was necessary.
https://github.com/odoo/o-spreadsheet/pull/2066/files#diff-b0adb6ac15e927159a8739ac1456275e9371da9b7397a6c87b363fad6db1feb5R192